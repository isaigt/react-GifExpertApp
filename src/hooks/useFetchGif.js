import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGif = ( categoria ) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs( categoria ).then( imgs => {
            setTimeout(() => {
                setState({
                    data: imgs,
                    loading: false
                });
            }, 1500);
        });
    }, [ categoria ]);

    return state;
}