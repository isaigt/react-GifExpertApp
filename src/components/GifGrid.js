import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoria }) => {
    const { data:images, loading } = useFetchGif( categoria );

    return (
        <>
            <h3 className="card animate__animated animate__fadeInDown">{ categoria }</h3>
            { loading && <h4 className="animate__animated animate__rotateInDownLeft">Cargando...</h4> }
            <div className="card-grid">
                    {
                        images.map( imagen => (
                            <GifGridItem key={ imagen.id } {...imagen} />
                        ))
                    }
            </div>
        </>
    );
}
