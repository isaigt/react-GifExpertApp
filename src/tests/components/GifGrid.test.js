import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGif } from '../../hooks/useFetchGif';
jest.mock('../../hooks/useFetchGif');

describe('Pruebas con el componente <GifGrid />', () => {
    
    test('debe hacer match con el snapshot', () => {
        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        });

        const categoria = 'One Punch';
        const wrapper = shallow( <GifGrid categoria={ categoria } /> );
        
        expect( wrapper ).toMatchSnapshot();
    });

    test('deben de mostrarse los elementos del GrifGridItem', () => {
        const imgs = [{
            id: '123456',
            title: 'Hola Mundo',
            url: 'https://localhost/holamundo.gif'
        }];

        useFetchGif.mockReturnValue({
            data: imgs,
            loading: false
        });
        
        const categoria = 'One Punch';
        const wrapper = shallow( <GifGrid categoria={ categoria } /> );

        //  expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h4').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( imgs.length );
    });
});
