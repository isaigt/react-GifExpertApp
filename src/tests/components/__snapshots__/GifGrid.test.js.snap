// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifGrid /> debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeInDown"
  >
    One Punch
  </h3>
  <h4
    className="animate__animated animate__rotateInDownLeft"
  >
    Cargando...
  </h4>
  <div
    className="card-grid"
  />
</Fragment>
`;
